File: api_nestjs_distributed_order_platform/src/application/use-case/client/create-client.use-case.ts
import { Inject, Injectable } from "@nestjs/common";
import type { ClientRepositoryInterface } from "src/application/port/client.repository";
import { ClientEntity } from "src/domain/entities/client.entity";
import { CreateClientInput } from "src/shared/interfaces/client.interface";
import { CLIENT_REPOSITORY } from "src/shared/tokens_nest/client.token";

@Injectable()
export class CreateClientUseCase {
    constructor(
        @Inject(CLIENT_REPOSITORY)
        private readonly clientRepository: ClientRepositoryInterface,
    ) {}

    async execute(input: CreateClientInput): Promise<ClientEntity> {
        const client = new ClientEntity(input.name, input.email, input.status);
        return this.clientRepository.create(client);
    }
}

File: api_java_distributed_order_platform/src/main/java/com/project/order/application/usecase/client/CreateClientUseCase.java
package com.project.order.application.usecase.client;

import com.project.order.application.port.ClientRepository;
import com.project.order.domain.model.Client;
import org.springframework.stereotype.Service;

@Service
public class CreateClientUseCase {

    private final ClientRepository clientRepository;

    public CreateClientUseCase(ClientRepository clientRepository) {
        this.clientRepository = clientRepository;
    }

    public Client execute(String name, String email) {
        Client client = Client.create(name, email);
        return clientRepository.create(client);
    }
}
